Dev env updates (#13296)
docker/docker.github.io: desktop/dev-environments.md

> **Note**
>

> > Known issue: With versions prior Desktop 3.6.0, when cloning a Git branch using `https://`, you must remove the `.git` suffix at the end of the URL.

Author: Should we remove this note totally now this is fixed ? (or put it at the bottom with other notes ?)

Reviewer: Perhaps something like

```suggestion
> Make sure the URL does not have a `.git` suffix if you are cloning a Git branch
> using `https://` and are using Docker Desktop prior to 3.6.0.
```

Yes, it would be good to keep "no longer" issues out of the main docs (after all, we'd like people to use a current version).

Perhaps add a "known issues" section in the v3.5 release notes for Mac and Windows, and move it there? https://github.com/docker/docker.github.io/blob/master/docker-for-mac/release-notes.md
https://github.com/docker/docker.github.io/blob/master/docker-for-windows/release-notes.md

And be sure to add a note to the 3.6 release notes that it's fixed.
3.6 has not been release yet though, so we need to wait for that (or combine this change with the release notes PR for that?)

That said; the example does not show a `.git` suffix in the `https://` example; do we need the note at all? (we could consider it a user error if they added `.git`?

Does the `git@` example also work if you remove the `.git` suffix? If so, perhaps remove the suffix there as well to prevent users accidentally adding one.

Reviewer: Wondering; nothing in the example is _pushing_ too GitHub. Should we remove all uses of `git@` and replace them with `https://` ? Doing so won't require users to set up ssh, and makes sure the tutorial "just works"

If we _do_ want to document how to work with ssh authentication, I think we should move that to a separate ("advanced") section, so that the "main" tutorial is simple to use, and gives the "it works!" experience.,  and worry about authentication later.

Author: Indeed, 3.6 has not been released yet. 
Makes sense to add a note in 3.6 release that this is fixed. I think we can then remove it from the main doc (esp on a preview, and more a side feature, not totally the main path)

Reviewer: @gtardif @rumpl any ideas on using `https://` in the examples on this page to simplify things? (see above)

Reviewer 2: I agree yes, I think I saw somewhere that people use https more than ssh when they use git.

Reviewer: Yes, `https://` is quite common, and has some advantages (mostly ease of use, and less chance of corporate firewalls blocking requests); https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols#_the_http_protocols

We should have some info on how to use dev-containers with `git` / `ssh` ("advanced" use), but if we don't need it for the first introduction, my vote would be to keep things easy for the user.

![Dev environment shared](/images/dev-env-shared.png){:width="700px"}

> Your team members need to open the **Create** Dialog, go to **Existing Dev Environment** tab, and paste this URL. Your Dev Environment now starts in the exact same state as you shared it!

Reviewer 3: ```suggestion
Your team members need to open the **Create** dialog, select the **Existing Dev Environment** tab, and then paste the URL. Your Dev Environment now starts in the exact same state as you shared it.
```

![Dev environment shared](/images/dev-env-shared.png){:width="700px"}

> Your team members just need to open the **Create** Dialog, go to **Existing Dev Environment** tab, and paste this URL. Your Dev Environment now starts in the exact same state as you shared it!

Reviewer 2: ```suggestion
Your team members need to open the **Create** Dialog, go to **Existing Dev Environment** tab, and paste this URL. Your Dev Environment now starts in the exact same state as you shared it!
```

  >
  > When cloning a Git repository using SSH, ensure you've added your SSH key to the ssh-agent. To do this, open a terminal and run `ssh-add <path to your private ssh key>`.

> 1. Click **Create** to open the create dialog box. Then copy `git@github.com:dockersamples/compose-dev-env.git` and add it to the Repository URL field.

Reviewer 3: ```suggestion
1. Click **Create** to open the **Connect to a Git or Docker Hub project** dialog. Then, copy `git@github.com:dockersamples/compose-dev-env.git` and add it to the **Repository URL** field on the **Remote** tab.
```

>
> To get involved with the discussion on how we are going to implement this as part of Compose, join the **#docker-dev-environments** channel in the [Docker Community Slack](https://dockercommunity.slack.com/messages){:target="_blank" rel="noopener" class="_"}, or let us know your feedback by creating an issue in the [Dev Environments](https://github.com/docker/dev-environments/issues){:target="_blank" rel="noopener" class="_"} GitHub repository.

## Start a Dev Environment from a local folder

> You can also start a Dev Environments from local code you have on your machine.

Reviewer 3: ```suggestion
You can also start a Dev Environment from local code on your machine.
```

> 1. Click **Create** to open the create dialog box. Then move to **Local folder** tab, and click **Select directory** to select the root of the code you want ot work on.

Reviewer 3: ```suggestion
1. Click **Create** to open the **Connect to a Git or Docker Hub project** dialog. Select the **Local** tab, and click **Select directory** to open the root of the code that you would like to work on.
```

2. Now, click **Create**.

    This creates a Dev Environment using your local folder, and bind-mounts your local code in the Dev Environment. Finally, it opens VS Code inside the Dev Environment container.

> **Note**
>

> > When using a local folder for a Dev Environment, file changes are synchronized between your Dev Environment container and your local files. This can affect performances inside the container, depending on the number of files in your local folder and operations performed in the container.

Reviewer 3: ```suggestion
> When using a local folder for a Dev Environment, file changes are synchronized between your Dev Environment container and your local files. This can affect the performance inside the container, depending on the number of files in your local folder and the operations performed in the container.
```

>
> To get involved with the discussion on how we are going to implement this as part of Compose, join the **#docker-dev-environments** channel in the [Docker Community Slack](https://dockercommunity.slack.com/messages){:target="_blank" rel="noopener" class="_"}, or let us know your feedback by creating an issue in the [Dev Environments](https://github.com/docker/dev-environments/issues){:target="_blank" rel="noopener" class="_"} GitHub repository.

## Start a Dev Environment from a local folder

You can also start a Dev Environments from local code you have on your machine.

1. Click **Create** to open the create dialog box. Then move to **Local folder** tab, and click **Select directory** to select the root of the code you want ot work on.
2. Now, click **Create**.

>     This creates a Dev Environment using your local folder, and bind-mount your local code in the Dev Environment. Finally, it opens VS Code inside the Dev Environment container.

Reviewer 2: ```suggestion
    This creates a Dev Environment using your local folder, and bind-mounts your local code in the Dev Environment. Finally, it opens VS Code inside the Dev Environment container.
```
